boards: arm: nucleo_u575zi_q: Added jlink runner

scottlaboe opened the pull request:
Segger provides a method for updating a Nucleo evaluation board's firmware to JLink firmware as documented [here](https://www.segger.com/products/debug-probes/j-link/models/other-j-links/st-link-on-board/). I have tested this with the Nucleo L552ZE_Q board and confirmed the runner functions properly.

scottlaboe committed:
Added jlink runner for STM32 Nucleo L552ZE_Q board

Signed-off-by: Scott Laboe <scottyl2.718@gmail.com>

erwango commented:
Thanks for this contribution, please fix compliances issues:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/5070844625/jobs/9119516973?pr=58244#step:9:24

scottlaboe committed:
boards: arm: nucleo_u575zi_q: Added jlink runner

Segger provides a method for updating a Nucleo evaluation
board's firmware to JLink firmware. I have tested this with the
Nucleo L552ZE_Q board and confirmed the runner functions
properly for flashing and debugging target.

Signed-off-by: Scott Laboe <scottyl2.718@gmail.com>

scottlaboe committed:
: boards: arm: nucleo_u575zi_q: Added jlink runner

Segger provides a method for updating a Nucleo evaluation
board's firmware to JLink firmware. I have tested this with the
Nucleo L552ZE_Q board and confirmed the runner functions
properly.

scottlaboe committed:
Merge branch 'main' of github.com:scottlaboe/zephyr

scottlaboe committed:
board: arm: nucleo: Added flash partitioning

Provided partitions for two different Nucleo boards
that I have on hand to allow easier integration with
mcuboot. The nucleo_l552ze_q board is configured
with a single slot due to its limited memory.

Signed-off-by: Scott Laboe <scottyl2.718@gmail.com>

scottlaboe commented:
Good lord I've really messed this up. I didn't realize I was pushing changes to this still with the last few commits. I'm just going to close this.
